File: src/main/java/io/dplabs/bookshelf/models/Collection.java
package io.dplabs.bookshelf.models;

import lombok.Data;

import javax.persistence.Entity;
import javax.validation.constraints.NotBlank;

@Entity(name = "collections")
@Data
public class Collection extends BaseEntity {
    @NotBlank(message = "Name is mandatory")
    private String name;

    private String description;

    private String link;
}

File: backend/src/repository/entities/collection.ts
import { Column, Entity, OneToMany } from 'typeorm';
import { BaseEntity } from './base-entity';
import { Book } from './book';

@Entity({ name: 'collections' })
export class Collection extends BaseEntity {
  @Column()
  name: string;

  @OneToMany(() => Book, (book) => book.category)
  books: Book[];

  @Column({ nullable: true })
  link: string;

  @Column({ nullable: true })
  description: string;
}
